state/protocol/util.go

```diff
@@ -69,3 +70,26 @@ func IsSporkRootSnapshot(snapshot Snapshot) (bool, error) {
 	}
 	return true, nil
 }
+
+// FindGuarantors decodes the signer indices from the guarantee, and finds the guarantor identifiers from protocol state
+func FindGuarantors(state State, guarantee *flow.CollectionGuarantee) ([]flow.Identifier, error) {
```

Comment:
would be good to document expected error returns here:
```suggestion
// Expected Error returns during normal operations:
//  * signature.ErrIncompatibleBitVectorLength indicates that `signerIndices` has the wrong length
//  * signature.ErrIllegallyPaddedBitVector is the vector is padded with bits other than 0
//  * signature.ErrInvalidChecksum if the input is shorter than the expected checksum contained therein
func FindGuarantors(state State, guarantee *flow.CollectionGuarantee) ([]flow.Identifier, error) {
```